### ISSUE TITLE
feature request - compare two folders

### ISSUE THREAD

**Issue by silkyclouds** [coderaiser/cloudcmd#220]
First of all thank you for the tool. I love it ! 

I use it everyday to copy content from my local server to a mounted rclone drive.

It would be really helpful for me (and other users I believe) to have a compare feature added to cloud commander, which would act like the compare feature in flashfxp.

so when the compare button has been pressed, only the non matching folders will remaing on each side of the screen. this allowing to quickly transfer ONLY what's on present on the destination drive. 

I hope this is clear and that you guys will think about implementing it in the future ! 

Thanks !

**Comment by silkyclouds** [comment]
Hey there,

Well should it be an icon or a hot key doesn’t really make a difference. The idea is simply to have a way to compare folder A and folder B. after pressing the button (or the hot key) only non-matched items will be visible on both sides of the screen ;) 

Thanks for your fast answer by the way ! :)

**Comment by coderaiser** [comment]
This could be solved with help of [User Menu](https://cloudcmd.io#user-menu).
All you need is create file `~/.cloudcmd.menu.js` with contents:

```js
module.exports = 
    'D - Compare directories': async ({DOM}) => {
        const {
            CurrentInfo,
            getFilenames,
            getCurrentByName,
            selectFile,
        } = DOM;
        
        const {
            files,
            filesPassive,
            panel,
            panelPassive,
        } = CurrentInfo;

        const names = getFilenames(files);
        const namesPassive = getFilenames(filesPassive);

        const selectedNames = compare(names, namesPassive);
        const selectedNamesPassive = compare(namesPassive, names);

        selectNames(selectedNames, panel, {
            selectFile,
            getCurrentByName,
        });

        selectNames(selectedNamesPassive, panelPassive, {
            selectFile,
            getCurrentByName,
        });
    },
}

function selectNames(names, panel, {selectFile, getCurrentByName}) {
    for (const name of names) {
        const file = getCurrentByName(name, panel);
        selectFile(file);
    }
};

function compare(a, b) {
    const result = [];

    for (const el of a) {
        if (b.includes(el))
            continue;

        result.push(el);
    }

    return result;
}
```

Start `Cloud Commander` with: `cloudcmd --user-menu --save`. Then press `F2` and `D`.

You can find more user menu examples in [User Menu Cookbook](https://github.com/coderaiser/cloudcmd/wiki/User-Menu-Cookbook#compare-directories).

Is it works for you?

**Comment by coderaiser** [comment]
Landed in [v12.5.0](https://github.com/coderaiser/cloudcmd/releases/tag/v12.5.0) as one of Default User Menu items 🎉.

**Comment by silkyclouds** [comment]
Hey this is great thank you very much!!!

Raphael
